include/nbl/builtin/hlsl/sort/counting.hlsl

```diff
@@ -104,9 +104,10 @@ struct counting
             sdata.workgroupExecutionAndMemoryBarrier();
 
             // no aliasing anymore
-            const uint32_t val = vid < KeyBucketCount ? sdata.get(vid) : 0;
-            sum = inclusive_scan(val, sdata);
-            histogram.atomicAdd(vid, sum);
+            if (vid < KeyBucketCount) {
+                sum = inclusive_scan(sdata.get(vid), sdata);
```

Comment:
here's the thing, `inclusive_scan` uses workgroup barriers, so it needs to be outside of non-uniform control flow

the `sdata.get()` you can do speculatively (OOB read unlikely to mess up your GPU), its only the atomic add (a write) that needs to be protected against OOB